perch/model.py

```diff
@@ -201,7 +265,18 @@ def update(self, user, **kwargs):
                 err = 'User is not authorised to update this resource'
                 raise exceptions.Unauthorized(err)
 
-        read_only_fields = set(kwargs.keys()) & set(self._resource.keys()) & self._read_only
+        fields_to_update = set(kwargs.keys()) & set(self._resource.keys())
+
+        state_field_update = fields_to_update & {'state'}
+        if state_field_update and \
+                not (yield self.validate_state_transition(user, self._resource.get('state'), kwargs['state'], **kwargs)):
+            err = [{
+                'field': 'state',
+                'message': 'Cannot transition from {} state to {} state'.format(self.state.name, kwargs['state'])
```

Comment:
I think this would be more readable if you wrap the long line using brackets, e.g.

```python
err = [{
    'field': 'state',
    'message': ('Cannot transition from {} state to {} state'
                .format(self.state.name, kwargs['state']))
}]
```